src/ComponentInstaller.php

```diff
@@ -187,12 +191,89 @@ public function onPostPackageInstall(PackageEvent $event)
                 return $injectors;
             }, new Collection([]))
             // Inject modules into configuration
-            ->each(function ($injector, $module) use ($name, $packageTypes) {
+            ->each(function ($injector, $module) use ($name, $packageTypes, $applicationModules) {
+                $injector->setApplicationModules($applicationModules);
                 $this->injectModuleIntoConfig($name, $module, $injector, $packageTypes);
             });
     }
 
     /**
+     * Find all Module classes in the package and include them.
+     * Module classes is used later
+     * @see \Zend\ComponentInstaller\Injector\AbstractInjector::injectAfterDependencies
+     * - to get package dependencies - module method `getModuleDependencies`
+     * and add component in a correct order on the module list.
+     *
+     * It works currently with PSR-0 and PSR-4 autoloaded packages.
+     *
+     * @param PackageInterface $package
+     * @return void
+     */
+    private function includeModuleClasses(PackageInterface $package)
+    {
+        $installer = $this->composer->getInstallationManager();
+        $packagePath = $installer->getInstallPath($package);
+
+        $autoload = $package->getAutoload();
+        foreach ($autoload as $type => $map) {
+            foreach ($map as $namespace => $path) {
+                switch ($type) {
```

Comment:
One more thing we may want to look for is the `files` type, as some modules will have `Module.php` in that array (as an example, all Apigility classes did that before we updated them; most zfc packages do as well). In most cases, when updated to expose themselves to zend-component-installer, they will likely be using psr-4, but something to consider as an edge case.